### DISCUSSION TITLE
Update feature when

### DISCUSSION THREAD

**Discussion by Kokie24** [OctoD/godot-gameplay-systems#50] [General]
I'm so worry that you are not update the feature on Godot Gameplay System now.

When do you update the source code and feature?

**Comment by OctoD** [comment]
Hi, I am working on a private repo for a full rewrite.

The new gdextension will be written in C++ for better performace and editor interoperability (an editor main view for the addon is going to be added).

All the existing features will stay, but their api will change a bit to make development easier, faster and to give more room to different use cases.

There will be a also a new feature which will close #47.

Thank you!

**Reply by appdevnick** [reply]
This is great news, I honestly love what you've put together so far and have gotten good use out of it. Thank you for your work!

**Reply by OctoD** [reply]
Thank you for you kind words! Since I finished to move to my new home I will start working consistently on the addon parts from this week, I will rush the port asap.

**Comment by TheYellowArchitect** [comment]
Some months have passed, but know that I and many other devs are looking forward to this!
The only reason devs like me hold from using it, is because it's worth it to wait the GDExtension. Performance matters, otherwise it is costly to import like that

Edit: https://github.com/OctoD/godot_gameplay_attributes released 4 days ago, is it in GDExtension?

**Reply by OctoD** [reply]
Hello! Yes it is, it's a c++ extension for godot, the next to be released will be the ability system, then the others. 

All these extensions will be bundled and glued into this repo when all of those will be ported back to gdextension.

**Reply by TheYellowArchitect** [reply]
Impressive, sounds like your plugin will cover most multiplayer games, and many classic RPG ones!
I will test godot gameplay attributes next week :+1: 

Slightly off-topic: Know that this wiki/docs page you had posted in the past https://github.com/OctoD/godot-gameplay-systems/blob/main/docs/ability-system.md is instant download for many godot developers (it's also what made me interested to this plugin)
If I were to advertise/promote this plugin, I would spam that page. To make any ability system through the simplicity of tags is attractive to every developer :)
